### ISSUE TITLE
pwdLastSet not working

### ISSUE THREAD

**Issue by mawvdrunen** [Adldap2/Adldap2#733]
- Adldap2 Version: 6.0
- LDAP Type: Active Directory
- PHP Version: 7.3.10

### Description:
I'm trying to get the timestamp of the last time the user changed their password. I tried
`$timestamp = $user->getFirstAttribute('pwdLastSet');`, but this results in `NULL`. Am I misunderstanding this concept? Is there maybe a different method to get the last time the used changed the password?

**Comment by stevebauman** [comment]
Hi @mawvdrunen,

When you dump all of the users attributes, do you see the `pwdlastset` attribute there?

```php
$user = $provider->search()->users()->find('jdoe');

var_dump($user->getAttributes());
```

**Comment by mawvdrunen** [comment]
Hi @stevebauman,

I'm not entirely sure if I'm doing it correctly, since the code you mention throws an exception "Call to a member function getAttributes()". Just to be sure, my code looks as follows:

```
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Adldap\AdldapInterface;

class LoginController extends Controller
{
    protected $ldap;

    public function __construct(AdldapInterface $ldap){
        $this->ldap = $ldap;
    }

    public function login(Request $request){
        $test = $this->ldap->search()->users()->find('username');
        var_dump($test->getAttributes());
    }
}
.
.
.
```

**Comment by stevebauman** [comment]
Does a user exist in your domain named `username`? That’s what you’re searching for in the example above...

**Comment by mawvdrunen** [comment]
While running the code I replaced username with an actually existing user. For the snippet I pasted above I left username.

**Comment by stevebauman** [comment]
Okay, it seems it's not locating the user you're searching for, as it's returning `null`.

Use `findByDnOrFail()` instead and insert a users full distinguished name:

```
$test = $this->ldap->search()->users()->findByDnOrFail('cn=john doe,dc=acme,dc=org');

var_dump($test->getAttributes());
```

**Comment by mawvdrunen** [comment]
This keeps returning `null`, however `$test = $this->ldap->search()->where('uid', '=', 'mvdrunen')->first();` does return a user. When I then `var_dump($test->getAttributes());` it displays the attributes, but pwdLastSet is not there.

**Comment by stevebauman** [comment]
> This keeps returning `null`

Are you sure you're inserting the correct distinguished name for a user from your directory?

If `pwdLastSet` is not there, then your server is not returning the attribute. Adldap2 by default selects all attributes to be returned. What user are you binding to your directory with? A standard user or an administrator?

The user you bind to your directory with must have read access to the property for it to be returned and visible.

**Comment by mawvdrunen** [comment]
> Are you sure you're inserting the correct distinguished name for a user from your directory?
Absolutely. I looked it up using an LDAP browser and used the exact distinguished name.
> The user you bind to your directory with must have read access to the property for it to be returned and visible.
I'll try getting different credentials from net Network Admin to try it out. Would a user always be able to access their own pwdLastSet property? So if I'd run `$this->ldap->auth()->attempt($username, $password, true);` first and then search for the user just authenticated and select the attribute it should be returned if the attribute is there?

**Comment by mawvdrunen** [comment]
I received admin credentials for the LDAP server and it's still not shown. Could it be that the property is just not stored because of configuration of the LDAP server?
